src/App.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Navigate, Route, Routes } from 'react-router-dom';
+import { CoursesPage } from './components/CoursesPage/CoursesPage';
+import { EnglishPage } from './components/EnglishPage/EnglishPage';
+import { HomeTaskPage } from './components/HomeTaskPage/HomeTaskPage';
+import { NotFoundPage } from './components/NotFoundPage';
+import { NotImplementedPage } from './components/NotImplementedPage';
+import { CoursePage } from './components/CoursePage/CoursePage';
+import './App.scss';
+
+export const App: React.FC = () => {
+  return (
+    <div className="App">
+      <Routes>
+        <Route
+          path="*"
+          element={<NotFoundPage />}
+        />
+
+        <Route path="/courses">
+          <Route
+            index
+            path="/courses"
+            element={<CoursesPage />}
+          />
+          <Route
+            path="/courses/:slug"
+            element={<CoursePage />}
+          />
+        </Route>
+
+        <Route path="/" element={<Navigate to="/courses" replace />} />
```

Comment:
```HTML 
path="/courses"
```
This is useless, it  will be better to remove this string. Nesting routes will still allow you to show all courses on the "/courses" url after this changes.

```HTML
path="/courses/:slug"
```
Replace this with the following code:
```HTML
path=":slug"
```
After those improvements you will still have ability to view the course by the "/courses/someId" link.